examples/arithmetic/example.swift

```diff
@@ -0,0 +1,4 @@
+import arithmetic
+
+print("Alrighty, let's do some arithmetic FROM RUST, IN SWIFT!")
+print("2 + 3 = \(add(a: 2, b: 3, overflow: .saturating))")
```

Comment:
Huh. I don't know much about swift so I want to double-check what I read a bit about online: is this thing where you have to name all the arguments just an expected convention in the swift world, or are we doing something that means you have to explicitly name them in cases where the caller might not expect to need to?

Reply:
I'm glad you pointed this out!

I think, as of [this RFC](https://github.com/apple/swift-evolution/blob/master/proposals/0046-first-label.md), all parameters need a label by default, unless you use `_`. If you omit the label in the declaration, like the generated code does, it'll be the same as the argument name. But I'm not sure if we can come up with a generic convention. There are some [guidelines](https://swift.org/documentation/api-design-guidelines/#argument-labels) for labels (and a great post with some more [rules of thumb](https://forums.swift.org/t/when-to-use-argument-labels-a-new-approach/1289/7)), but it doesn't seem like something we can infer in the general case. It only takes a few cases for "good enough" to produce bizarro results—like `intersection(a, ln2: b)`.

I was thinking "no label if only one argument, labels if more than one" would be a general enough rule, but even that's not the case. [This thread](https://forums.swift.org/t/when-to-use-argument-labels-a-new-approach/1289) has good counterexamples (`a.dismiss(animated: b)`; also, something like `sort(stable: Bool)`). On the other hand, with a function like `zip(a, b)` or `intersection(a, b)`, the names are redundant.

I think the answer might be in adding smarts to the IDL syntax. What if we added a special annotation for labels, or let you specify labels in declarations with a special syntax (like `Point? intersection(Line ln1 labeled _, Line ln2 labeled _)`, as an example)...or let you to omit argument names (so `Point? intersection(Line, Line)` would imply no labels, while `Point? intersection(Line ln1, Line ln2)` would emit labels)?

On the other hand, none of the other languages we target support this. Python and Kotlin have keyword/named arguments, but they're the same as the argument names. I guess we could slurp up labeled arguments into `**kwargs` in Python, or options objects in JS, but _why_, when you can pass a `record` instead? That's a whole lotta work to support a syntax that's only special in Swift.

Another idea is, what if no labels at all? They do help with clarity, but...no labels would match our other generated bindings.

Reply:
I have seen adding a naming conventions in the IDL work okayish.

e.g. count the number of suffix words in the function name, and use those to name the arguments; if you run out of suffix words, use `_`.

```idl
Point? intersection_with(Line line1, Line line2)
```

generating

```swift
func intersection(_ line1: Line, with line2: Line) -> Point?
```

But there're a tons of ways this will produce rubbish results, particularly if we're mixing other naming conventions.

I like your annotation idea but I have no preference of syntax.

```idl
@argumentLabels("_", "with")
Point? intersection(Line line1, Line line2)
```

But I think the reason for this comment is around collisions. 

There may be nothing to do for now, but it's adjacent to this problem: swift and kotlin both have different ways of defining function signatures (method name + argument labels vs method name + argument types).